sounddevice.py

```diff
@@ -2262,13 +2262,68 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
```

Comment:
What if somebody was just wrapping one of the sounddevice functions or classes and wanted to catch a certain exception and re-raise it with their own message? How would they know that the error code argument is to come second? Anyway, that's the use case I had imagined. I'm not sure that anybody would actually need that functionality.

On a related note, reraising a PortAudioError that is a host api error has some unintended repercussions. It would cause GetLastHostError to get called again and I really have no idea if there is even well defined behavior if this happens and much less so if it is the behavior we would want. I had initially left the host error formatting and the call to GetLastHostError inside of `_check` for this very reason.